src/main/java/seedu/clinic/logic/parser/EditCommandParser.java

```diff
@@ -203,4 +181,21 @@ private EditDescriptor parseGeneralDetails(EditDescriptor editDescriptor, Argume
 
         return editDescriptor;
     }
+
+    private ParseException checkInvalidArguments(Prefix prefix, ArgumentMultimap argMultimap) {
+        if (argMultimap.getValue(prefix).get().contains("/")) {
+            return new ParseException(MESSAGE_INVALID_PREFIX + "\n" + EditCommand.MESSAGE_USAGE);
+        }
+        if (argMultimap.getValue(prefix).get().split("\\s+").length != 1) {
+            return new ParseException(String.format(MESSAGE_INVALID_USAGE, EditCommand.MESSAGE_USAGE));
+        }
+        if (prefix.equals(PREFIX_TYPE)) {
+            return new ParseException(String.format(MESSAGE_INVALID_TYPE_EDIT, EditCommand.MESSAGE_USAGE));
+        } else if (prefix.equals(PREFIX_INDEX)) {
+            return new ParseException(MESSAGE_INVALID_INDEX + "\n" + EditCommand.MESSAGE_USAGE);
+        } else {
+            assert prefix.equals(PREFIX_PHONE) : "The prefix here should be of Phone type!";
+            return new ParseException(Phone.MESSAGE_CONSTRAINTS + "\n" + EditCommand.MESSAGE_USAGE);
+        }
```

Comment:
May I know the intention of this method? I though the checking of invalid argument is done during the class construction itself, for example:
 ```
public Address(String address) {
        requireNonNull(address);
        checkArgument(isValidAddress(address), MESSAGE_CONSTRAINTS);
        value = address;
    }
```

Reply:
I can remove this if we want to just show a more generic error message instead. For instance, if someone entered p/99999999 z/testing, using the check in class construction will show an error message that the phone number is not valid. However, this is not the case as the number attached to prefix p is valid, except an incorrect prefix was added at the end, resulting in the check for isValidPhone to fail. Same for checks where p/99999999 testing when user enters random string value behind a prefix (or will this case be treated as wrong input?)